web/api/v1/api.go

```diff
@@ -76,17 +77,19 @@ const (
 type errorType string
 
 const (
-	errorNone          errorType = ""
-	errorTimeout       errorType = "timeout"
-	errorCanceled      errorType = "canceled"
-	errorExec          errorType = "execution"
-	errorBadData       errorType = "bad_data"
-	errorInternal      errorType = "internal"
-	errorUnavailable   errorType = "unavailable"
-	errorNotFound      errorType = "not_found"
-	errorNotAcceptable errorType = "not_acceptable"
+	ErrorNone          errorType = ""
+	ErrorTimeout       errorType = "timeout"
+	ErrorCanceled      errorType = "canceled"
+	ErrorExec          errorType = "execution"
+	ErrorBadData       errorType = "bad_data"
+	ErrorInternal      errorType = "internal"
+	ErrorUnavailable   errorType = "unavailable"
+	ErrorNotFound      errorType = "not_found"
+	ErrorNotAcceptable errorType = "not_acceptable"
 )
 
+type ErrorTypeToStatusCode map[errorType]func(error) int
```

Comment:
Is it valid to export this and leave `errorType` unexported?  How would someone implement it?

Reply:
Although the type is not exposed, the enums themselves have been changed to be exposed. So you would use it like this:
```
api := API{
        ...
	errorTypeToStatusCode: ErrorTypeToStatusCode{
		ErrorExec: func(err error) int {
			if strings.Contains(err.Error(), "too many requests") {
				return 429
			}
			return 422
		},
	},
}
```